controller/network/routesender.go

```diff
@@ -106,6 +110,7 @@ attendance:
 
 					if status.r == tr {
 						strategy.NotifyEvent(xt.NewDialFailedEvent(terminator))
```

Comment:
I feel a refactor coming on for a lot of these kinds of internals. This `routeSender.route`, `network.Run`, etc., should probably all have `interface{}` types that define extension points where we can insert these kinds of strategies (handlers).

I suspect we're probably going to end up wanting to clean up a bunch of these internals in this way. Eventually our `controller` and `router` stacks probably get built through some big IoC-style factory.